thentos-core/src/Thentos.hs

```diff
@@ -149,3 +153,18 @@ createDefaultUser st (Just (getDefaultUser -> (userData, roles))) = do
         if all isRight result
             then logger DEBUG $ "[ok]"
             else logger ERROR $ "failed to assign default user to roles: " ++ ppShow (UserId 0, result, user, roles)
+
+-- | Autocreate any services that are listed in the config but don't exist in the DB.
+autocreateMissingServices :: (db `Ex` DB) => ThentosConfig -> Action db ()
+autocreateMissingServices cfg = do
+    dieOnDuplicates
+    mapM_ (autocreateServiceIfMissing'P agent) allSids
+  where
+    -- Die with an error if the default "proxy" service ID is repeated in the "proxies" section
```

Comment:
Could you move this comment to the haddocks? It's fine to throw an error on startup when first reading the config file, but there's always the chance that this function will be used elsewhere, where it might not be okay.